pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -48,6 +48,78 @@ type ClusterList struct {
 	Items []Cluster `json:"items"`
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterVersion represents a version of OpenShift that can be, or is installed.
+type ClusterVersion struct {
+	// +optional
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +optional
+	Spec ClusterVersionSpec `json:"spec,omitempty"`
+	// +optional
+	Status ClusterVersionStatus `json:"status,omitempty"`
+}
+
+// ClusterVersionSpec is a specification of a cluster version that can be installed.
+type ClusterVersionSpec struct {
+	// +optional
+	YumRepositories []YumRepository `json:"yumRepositories,omitempty"`
+
+	// ImageFormat defines a format string for the container registry and images to use for
+	// various OpenShift components. Valid expansions are component (required, expands to
+	// pod/deployer/haproxy-router/etc), and version (v3.9.0).
+	// (i.e. example.com/openshift3/ose-${component}:${version}")
+	ImageFormat string `json:"imageFormat"`
+
+	VMImages VMImages `json:"vmImages"`
+}
+
+// ClusterVersionStatus is the status of a ClusterVersion. It may be used to indicate if the
+// cluster version is ready to be used, or if any problems have been detected.
+type ClusterVersionStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterVersionList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []ClusterVersion `json:"items"`
+}
+
+// VMImages contains required data to locate images in all supported cloud providers.
+type VMImages struct {
+	// +optional
+	AWSImages *AWSVMImages `json:"awsVMImages,omitempty"`
+	// TODO: GCP, Azure added as necessary.
+}
+
+// AWSVMImages indicates which AMI to use in each supported AWS region for this OpenShift version.
+type AWSVMImages struct {
+	AMIByRegion map[string]string `json:"amiByRegion"`
+}
+
+// YumRepository represents optional yum repositories to deploy onto all systems in the cluster.
+type YumRepository struct {
+	// Using lowercase for JSON as this matches the properties in the actual yum repo / openshift-ansible structure:
```

Comment:
This is probably not the best place for this comment as it will show up associated with the `ID` field. It seems better for it to be associated with the `YumRepository` type. But it might not be appropriate to show up for the end user at all.